File: python/344_Reverse-String.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        if len(s)==1: return
        ptr1, ptr2, tmp = 0, len(s)-1, 0

        while(ptr1<ptr2):
            
            if s[ptr1] != s[ptr2]:
                #swap
                tmp = s[ptr1]
                s[ptr1] = s[ptr2]
                s[ptr2] = tmp

            ptr1+=1
            ptr2-=1
        

File: c++/344_Reverse-String.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        reverse(s.begin(), s.end());
    }
};

class Solution {
public:
    void reverseString(vector<char>& s) {
        int n = s.size();
        for(int i = 0; i <= (n-1-i); ++i){
            swap(s[i], s[n-1-i]);
        }
    }
};